pkg/v1/tkg/client/upgrade_cluster.go

```diff
@@ -345,7 +345,11 @@ func (c *TkgClient) upgradeAddonPreNodeUpgrade(regionalClusterClient clusterclie
 	// tanzu-addons-manager and tkr-controller only runs in management cluster
 	if isRegionalCluster {
 		addonsToBeUpgraded = append(addonsToBeUpgraded,
-			"addons-management/tanzu-addons-manager", "tkr/tkr-controller", "addons-management/core-package-repo")
+			"addons-management/tanzu-addons-manager",
+			"tkr/tkr-controller",
+			"addons-management/core-package-repo",
+			"packages/management-package-repo",
```

Comment:
These are going to be upgraded pre node upgrade if that is what you are aiming for /cc @blc1996

Reply:
what do you think would be the right thing to do pre or post for featuregates?

Reply:
`upgradeAddonPreNodeUpgrade` happens before KubeadmControlPlane and MachineDeployment upgrades.
`upgradeAddonPostNodeUpgrade` happens after those 2 upgrades.
I'll leave that decision to you, you can consider from the aspect of compatibility.

Reply:
for featuregates management package, i'm fine whenever it upgrades, but for the other management packages I'm not sure.. @vijaykatam when would you recommend upgrading the management packages in general?

Reply:
let's upgrade the management package repo before node upgrade and revisit later if we have a good reason